script.js

```diff
@@ -0,0 +1,235 @@
+      return modal.autoPlay;
+    },
+
+    setAutoPlay: function (bool) {
+      modal.autoPlay = bool;
+    },
+
+    getSLIDE_TIMEOUT: function () {
+      return  modal.SLIDE_TIMEOUT;
+    }
+ };
+
+ /*------ V I E W ------*/
+ var viewSlide = {
+    init: function () {
+
+      this.slideElem = document.querySelector('.slideshow__slide');
+      this.slideTitleElem = document.querySelector('.slideshow__title');
+      this.slideImgElem = document.querySelector('#slideshow__img');
+      this.slideDescriptionElem = document.querySelector('.slideshow__paragraph');
+      this.slideReferenceElem = document.querySelector('.slideshow__reference');
+
+      this.slideTitleElem.addEventListener('click', function () {
+        viewSlide.slideReferenceElem.style.display = 
+              viewSlide.slideReferenceElem.style.display === 'none'? 'block' : 'none';
+      })
+
+      this.render();
+    },
+
+    render: function () {
+      var currentSlide = controler.getCurrentSlide;
+      this.slideTitleElem.firstElementChild.textContent = modal.currentSlide.title;
+      this.slideImgElem.src = modal.currentSlide.imgSrc;
+      this.slideDescriptionElem.firstElementChild.textContent = modal.currentSlide.description;
+      this.slideReferenceElem.textContent = modal.currentSlide.reference;
+      this.slideImgElem.alt = modal.currentSlide.reference;
+    },
+
+    autoRender: function () {
+      var slide = controler.getNextSlide();
+      controler.setCurrentSlide(slide);
+      viewSlide.render();
+      viewOptionBar.render();
+    }
+ };
+
+ var viewThumbnail = {
+  init: function () {
+    this.elemThumbnails = document.querySelector('.slideshow__thumbnails');
+
+    this.render();
+  },
+
+  render: function () {
+    var slides = controler.getSlides();
+
+    for(var i = 0; i < slides.length; i++){
+      slide = slides[i];
+
+      elemDiv = document.createElement('div');
+      elemDiv.classList.add('slideshow__thumbnail-imgs');
+
+      elemImg = document.createElement('img');
+      elemImg.src = slide.imgSrc;
+      elemDiv.appendChild(elemImg);
+
+      elemDiv.addEventListener('click', (function (fakeSlide) {
+        return function () {
+          controler.setCurrentSlide(fakeSlide);
+          viewSlide.render();
+        }
+      })(slide));
+
+      this.elemThumbnails.appendChild(elemDiv);
+    }
+  }
+
+ };
+
+ var viewOptionBar = {
+    init: function () {
+      this.elemButtonInfo = document.querySelector('.slideshow__button--info');
+      this.elemButtonThumbnails = document.querySelector('.slideshow__button--thumbnails');
+      this.elemCounter = document.querySelector('.slideshow__counter');
+      this.elemButtonPrev = document.querySelector('.slideshow__button--prev');
+      this.elemButtonPlay = document.querySelector('#slideshow__playBtn');
+      this.elemButtonNext = document.querySelector('.slideshow__button--next');
+      var intervalSlides;
+
+      this.elemButtonNext.addEventListener('click', function () {
```

Comment:
You could have all the logic to set all the listeners in a separate function called `setListeners`